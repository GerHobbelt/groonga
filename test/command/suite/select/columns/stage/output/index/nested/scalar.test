table_create Products TABLE_PAT_KEY ShortText

table_create Purchases TABLE_NO_KEY
column_create Purchases product COLUMN_SCALAR Products
column_create Purchases tag COLUMN_SCALAR ShortText

column_create Products purchases COLUMN_INDEX Purchases product

load --table Products
[
{"_key": "apple"},
{"_key": "banana"},
{"_key": "cacao"}
]

load --table Purchases
[
{"product": "apple",  "tag": "man"},
{"product": "banana", "tag": "man"},
{"product": "cacao",  "tag": "woman"},
{"product": "apple",  "tag": "many"}
]

select Products \
  --columns[tags].stage output \
  --columns[tags].flags COLUMN_VECTOR \
  --columns[tags].type ShortText \
  --columns[tags].value purchases.tag \
  --output_columns _key,tags
