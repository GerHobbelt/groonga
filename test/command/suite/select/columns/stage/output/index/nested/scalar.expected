table_create Products TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
table_create Purchases TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Purchases product COLUMN_SCALAR Products
[[0,0.0,0.0],true]
column_create Purchases tag COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
column_create Products purchases COLUMN_INDEX Purchases product
[[0,0.0,0.0],true]
load --table Products
[
{"_key": "apple"},
{"_key": "banana"},
{"_key": "cacao"}
]
[[0,0.0,0.0],3]
load --table Purchases
[
{"product": "apple",  "tag": "man"},
{"product": "banana", "tag": "man"},
{"product": "cacao",  "tag": "woman"},
{"product": "apple",  "tag": "many"}
]
[[0,0.0,0.0],4]
select Products   --columns[tags].stage output   --columns[tags].flags COLUMN_VECTOR   --columns[tags].type ShortText   --columns[tags].value purchases.tag   --output_columns _key,tags
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        3
      ],
      [
        [
          "_key",
          "ShortText"
        ],
        [
          "tags",
          "ShortText"
        ]
      ],
      [
        "apple",
        [
          "man",
          "many"
        ]
      ],
      [
        "banana",
        [
          "man"
        ]
      ],
      [
        "cacao",
        [
          "woman"
        ]
      ]
    ]
  ]
]
